corehq/apps/registry/fixtures.py

```diff
@@ -0,0 +1,116 @@
+from lxml.builder import E
+
+from casexml.apps.phone.fixtures import FixtureProvider
+from corehq import toggles
+from corehq.apps.app_manager.dbaccessors import get_apps_in_domain
+from corehq.apps.domain.models import Domain
+from corehq.apps.registry.helper import DataRegistryHelper
+from corehq.apps.registry.models import DataRegistry
+
+
+class RegistryFixtureProvider(FixtureProvider):
+    """Output fixtures for data registries"""
+    id = "registry"
+
+    def __call__(self, restore_state):
+        if not _should_sync(restore_state):
+            return []
+
+        apps = _get_apps(restore_state)
+        registry_slugs = {
+            module.search_config.data_registry
+            for app in apps
+            for module in app.get_modules()
+            if module.search_config.data_registry
+        }
```

Comment:
This'd be a good cache target - the result is tiny, fairly expensive to compute, and only varies by domain.  It'd be nice to cache the whole fixture xml, but I'm not sure we have a good framework for that, do we?

Reply:
the apps are cached so this shouldn't be that expensive should it?

We do have some caching that we use for lookup tables, I can take a look to see if we can use it at all and how we'd know when to bust the cache.

Reply:
I don't know - I was mostly thinking about the overhead of wrapping large apps.  Not sure how much of a contributor it'd be in practice, but yeah we can always look at example restores once this is live and see.

Reply:
https://dimagi-dev.atlassian.net/browse/USH-1278

Reply:
> I don't know - I was mostly thinking about the overhead of wrapping large apps. Not sure how much of a contributor it'd be in practice, but yeah we can always look at example restores once this is live and see.

Not that this makes it correct but the UCR fixture does the same thing: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/app_manager/fixtures/mobile_ucr.py#L122-L128

Reply:
Oh good point - there was a recent ticket about that fixture being surprisingly slow even with trivially small reports - I'll see if I can dig it up.